bundles/auth/org.eclipse.smarthome.auth.oauth2client/src/main/java/org/eclipse/smarthome/auth/oauth2client/internal/config/SymmetricKeyCipherConfiguration.java

```diff
@@ -0,0 +1,163 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.auth.oauth2client.internal.config;
+
+import java.io.IOException;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.util.Base64;
+import java.util.Dictionary;
+import java.util.Hashtable;
+import java.util.Map;
+
+import javax.crypto.Cipher;
+import javax.crypto.KeyGenerator;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.NullCipher;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+
+import org.eclipse.smarthome.core.auth.client.oauth2.CipherConfigurator;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceReference;
+import org.osgi.service.cm.Configuration;
+import org.osgi.service.cm.ConfigurationAdmin;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This is a symmetric key encryption service for encrypting the OAuth tokens.
+ *
+ * @author Gary Tse - Initial Contribution
+ *
+ */
+@Component(name = "org.eclipse.smarthome.auth.oauth2client.internal.config.SymmetricKeyCipherConfiguration", property = {
+        "BASE64IV=DCtiyX+oIVqR+vFK4OhFGA==" })
+public class SymmetricKeyCipherConfiguration implements CipherConfigurator {
+
+    private final Logger logger = LoggerFactory.getLogger(SymmetricKeyCipherConfiguration.class);
+
+    private static final String PID = "org.eclipse.smarthome.auth.oauth2client.internal.config.SymmetricKeyCipherConfiguration";
+    private static final String ENCRYPTION_ALGO = "AES/CBC/PKCS5Padding";
+    private static final String PROPERTY_KEY_ENCRYPTION_KEY_BASE64 = "ENCRYPTION_KEY";
+    private static final String PROPERTY_KEY_BASE64_IV = "BASE64IV";
+
+    private IvParameterSpec ivspec;
+    private SecretKey encryptionKey;
+
+    /**
+     * Load the encryption key, load the IV from properties
+     *
+     * @param properties
+     */
+    @Activate
+    public void activate(Map<String, Object> properties, BundleContext bundleContext) {
+        ServiceReference configAdminServiceRef = bundleContext.getServiceReference(ConfigurationAdmin.class.getName());
+        ConfigurationAdmin configAdminService = (ConfigurationAdmin) bundleContext.getService(configAdminServiceRef);
+        try {
+            Configuration configuration = configAdminService.getConfiguration(PID);
```

Comment:
use `configAdminService.getConfiguration(PID, "?");` An addition in spec that disables the location check.